libraries/rush-lib/src/cli/RushPnpmCommandLine.ts

```diff
@@ -0,0 +1,273 @@
+
+    try {
+      // Are we in a Rush repo?
+      let rushConfiguration: RushConfiguration | undefined = undefined;
+      if (RushConfiguration.tryFindRushJsonLocation()) {
+        // showVerbose is false because the logging message may break JSON output
+        rushConfiguration = RushConfiguration.loadFromDefaultLocation({ showVerbose: false });
+      }
+
+      NodeJsCompatibility.warnAboutCompatibilityIssues({
+        isRushLib: true,
+        alreadyReportedNodeTooNewError: !!options.alreadyReportedNodeTooNewError,
+        rushConfiguration
+      });
+
+      if (!rushConfiguration) {
+        throw new Error(
+          'The "rush-pnpm" command must be executed in a folder that is under a Rush workspace folder'
+        );
+      }
+
+      if (rushConfiguration.packageManager !== 'pnpm') {
+        throw new Error(
+          'The "rush-pnpm" command requires your rush.json to be configured to use the PNPM package manager'
+        );
+      }
+
+      if (!rushConfiguration.pnpmOptions.useWorkspaces) {
+        throw new Error(
+          'The "rush-pnpm" command requires the "useWorkspaces" setting to be enabled in rush.json'
+        );
+      }
+
+      const workspaceFolder: string = rushConfiguration.commonTempFolder;
+      const workspaceFilePath: string = path.join(workspaceFolder, 'pnpm-workspace.yaml');
+
+      if (!FileSystem.exists(workspaceFilePath)) {
+        terminal.writeErrorLine('Error: The PNPM workspace file has not been generated:');
+        terminal.writeErrorLine(`  ${workspaceFilePath}\n`);
+        terminal.writeLine(Colors.cyan(`Do you need to run "rush install" or "rush update"?`));
+        throw new AlreadyReportedError();
+      }
+
+      if (!FileSystem.exists(rushConfiguration.packageManagerToolFilename)) {
+        terminal.writeErrorLine('Error: The PNPM local binary has not been installed yet.');
+        terminal.writeLine('\n' + Colors.cyan(`Do you need to run "rush install" or "rush update"?`));
+        throw new AlreadyReportedError();
+      }
+
+      // 0 = node.exe
+      // 1 = rush-pnpm
+      const pnpmArgs: string[] = process.argv.slice(2);
+
+      RushPnpmCommandLine._validatePnpmUsage(pnpmArgs, terminal);
+
+      const pnpmEnvironmentMap: EnvironmentMap = new EnvironmentMap(process.env);
+      pnpmEnvironmentMap.set('NPM_CONFIG_WORKSPACE_DIR', workspaceFolder);
+
+      if (rushConfiguration.pnpmOptions.pnpmStorePath) {
+        pnpmEnvironmentMap.set('NPM_CONFIG_STORE_DIR', rushConfiguration.pnpmOptions.pnpmStorePath);
+      }
+
+      if (rushConfiguration.pnpmOptions.environmentVariables) {
+        for (const [envKey, { value: envValue, override }] of Object.entries(
+          rushConfiguration.pnpmOptions.environmentVariables
+        )) {
+          if (override) {
+            pnpmEnvironmentMap.set(envKey, envValue);
+          } else {
+            if ('undefined' === typeof pnpmEnvironmentMap.get(envKey)) {
```

Comment:
```suggestion
            if (undefined === pnpmEnvironmentMap.get(envKey)) {
```
The `typeof` trick is only needed when trying to access global variables that might not be declared.

Reply:
(Also, reading from left to right, it's slightly odd to say "if absent is Bob" versus "if Bob is absent". This is a stylistic opinion though, not an important principle.)

Reply:
I know the habit historically comes from languages like C and avoiding "The last C bug", but nowadays we rely on linters to catch that.

Reply:
Anecdotally, this is called a "[Yoda condition](https://eslint.org/docs/rules/yoda) by eslint